src/main/holochainManager.ts

```diff
@@ -553,60 +553,55 @@ export class HolochainManager {
     return happSha256;
   }
 
+  private createUiDirectory(uiDir: string, uiBytes: Array<number>): void {
+    fs.mkdirSync(uiDir, { recursive: true });
+
+    const hashes: Record<string, string> = {};
+    const zip = new AdmZip(Buffer.from(uiBytes));
+    zip.getEntries().forEach((entry) => {
+      const hash = crypto.createHash('sha256').update(entry.getData()).digest('hex');
+      hashes[entry.entryName] = hash;
+    });
+
+    this.integrityChecker.storeToSignedJSON(path.join(uiDir, 'hashes.json'), hashes);
+    zip.extractAllTo(path.join(uiDir, 'assets'));
+  }
+
+  private storeIcon(uiDir: string, icon: Uint8Array): void {
+    const iconPath = path.join(uiDir, 'icon');
```

Comment:
If it's always a .png file and we store it as such we might as well call it `icon.png` here maybe.